src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -143,8 +168,13 @@ public function process(ContainerBuilder $container)
 
                 if (isset($this->hotPathEvents[$args[0]])) {
                     $container->getDefinition($id)->addTag($this->hotPathTagName);
+                } elseif (isset($this->noPreloadEvents[$args[0]])) {
+                    ++$noPreload;
                 }
             }
+            if ($noPreload && \count($extractingDispatcher->listeners) === $noPreload) {
+                $container->getDefinition($id)->addTag($this->noPreloadTagName);
```

Comment:
What if a class is both a subscriber and tagged with additional `kernel.listener` tags ?

Reply:
Then we don't really care IMHO: it should be exceptional, thus not worth any extra complexity. And the outcome is minor anyway.